src/cc2olx/tools/video_upload.py

```diff
@@ -0,0 +1,253 @@
+        * new_csv_path: path to a new CSV file that will be created that includes results of the video uploads
+        * file_data: a mapping from filename to data about the file to be include in the new CSV,
+            namely the edx_video_id
+
+
+    """
+    # utf-8 doesn't work; why does latin-1 work?
+    # csv_file = open(args.csv_file_path, encoding='utf-8')
+    with open(str(old_csv_path), encoding='latin-1') as csv_file, open(str(new_csv_path), 'w') as new_csv_file:
+        reader = csv.DictReader(csv_file)
+
+        new_fieldnames = reader.fieldnames.copy()
+        new_fieldnames.append('Edx Id')
+
+        writer = csv.DictWriter(new_csv_file, new_fieldnames)
+        writer.writeheader()
+
+        for row in reader:
+            filepath = row['Relative File Path']
+            try:
+                data = file_data[filepath]
+            except KeyError:
+                print("The file {} is missing from the supplied directory. "
+                      "No video was uploaded.".format(filepath))
+                continue
+
+            new_row = row.copy()
+            new_row["Edx Id"] = data["edx_video_id"]
+
+            writer.writerow(new_row)
+
+
+def main():
+    access_token = get_access_token()
+    args = get_args()
+
+    root = Path(args.directory)
+    get_upload_link_url = GENERATE_UPLOAD_LINK_BASE_URL + args.course_id
+    files_data = {}
+
+    for dirpath, _, filenames in os.walk(args.directory):
+        for filename in filenames:
+            # construct filename this way to uniquely identify
+            # the file within the root directory
+            full_path = Path(dirpath, filename)
+            relative_path = full_path.relative_to(root)
+            extension = relative_path.suffix
+
+            if extension in VIDEO_EXTENSION_CONTENT_TYPES:
+                content_type = VIDEO_EXTENSION_CONTENT_TYPES.get(extension)
+
+                request_data = {
+                    'files': [{'file_name': filename, 'content_type': content_type}]
+                }
+
+                response = make_generate_upload_link_request(get_upload_link_url, request_data, filename, access_token)
+
+                try:
+                    data = response.json()
+                except ValueError:
+                    print("Unable to parse JSON for call to the Studio generate upload link API "
+                          "for file {}.".format(filename))
+
+                edx_video_id = None
+                upload_url = None
+
+                if 'files' in data and data['files']:
+                    file_data = data['files'][0]
+
+                    if 'edx_video_id' in file_data:
+                        edx_video_id = file_data['edx_video_id']
+
+                    if 'upload_url' in file_data:
+                        upload_url = file_data['upload_url']
+
+                # upload video to presigned url if we have one
+                if upload_url:
```

Comment:
Can we do some error handling before this logic? Raise and print an error if upload_url is `None`. Then raise a warning if the `edx_video_id` is None but continue to do the upload?

Reply:
I can add error handling. 

But, I don't think we should upload a video without an `edx_video_id`. In fact, I don't see how we could receive an `upload_url` but not an `edx_video_id` in the response, as the `edx_video_id` is just a UUID that we generate [here](https://github.com/edx/edx-platform/blob/8866cc0e231b2bd4a161be75417c64e198f8d24f/cms/djangoapps/contentstore/views/videos.py#L754). The edx video ID is what will allow the second step of this tool to convert the Common Cartridge videos into video xBlocks.

I'd prefer just to print an error and skip the video if either the `upload_url` is missing or the `edx_video_id` is missing.